core/formatter.cpp

```diff
@@ -594,6 +595,26 @@ class Unparser {
  * The rest of this file contains transformations on the ASTs before unparsing. *
  ********************************************************************************/
 
+/** As a.push_back(elem) but preserves constraints.
```

Comment:
It's the same as in the function below, I am referring to it in "See concat_fodder". It's short, so maybe duplicating it won't be so bad. Or maybe there is a better place to put these constraints?